syntax = "proto2";

enum FluxMagic {
	MAGIC = 0x466c7578;
}

enum FluxFileVersion {
	VERSION_1 = 1;
	VERSION_2 = 2;
}

message TrackFluxProto {
	optional int32 track = 1;
	optional int32 head = 2;
	repeated bytes flux = 3;
}

message FluxFileProto {
	optional int32 magic = 1;
	optional FluxFileVersion version = 2;
	repeated TrackFluxProto track = 3;
	optional double rotational_period_ms = 4;
}

